### PR TITLE
Add all passing chip-repl yaml tests to CI along with some fixes required

### PR DESCRIPTION
**Pull request by tehampson**
This will prevent future breakages like some recently PR that contained a few small issues

### CODE REVIEW

File: scripts/tests/chiptest/yamltest_with_chip_repl_tester.py [hunk 1]
```diff
@@ -88,28 +88,31 @@ def main(setup_code, yaml_path, node_id):
...
+                if test_action is not None:
+                    response = runner.execute(test_action)
+                    decoded_response = runner.decode(response)
+                    post_processing_result = test_step.post_process_response(decoded_response)
+                    if not post_processing_result.is_success():
+                        raise Exception(f'Test step failed {test_step.label}')
+                else:
+                    raise Exception(f'Failed to encode test step {test_step.label}')
+        except Exception:
+            exit(-2)
```
Comment (andy31415) [inline scripts/tests/chiptest/yamltest_with_chip_repl_tester.py:115]: should we print a traceback here to debug? I am afraid exit will lose the exception message and stack (or is that not true?)

File: scripts/py_matter_yamltests/matter_yamltests/pics_checker.py [hunk 2]
```diff
@@ -52,7 +52,9 @@ def check(self, pics) -> bool:
 
         self.__expression_index = 0
         tokens = self.__tokenize(pics)
-        return self.__evaluate_expression(tokens, self.__pics)
+        if self.__pics is not None:
+            return self.__evaluate_expression(tokens, self.__pics)
+        return True
```
Comment (tehampson) [inline scripts/py_matter_yamltests/matter_yamltests/pics_checker.py:57]: Pretty sure this should actually be false

Comment (vivien-apple) [reply scripts/py_matter_yamltests/matter_yamltests/pics_checker.py:57]: I have done a PR with a test for the case where no pics file is used.

File: scripts/py_matter_yamltests/matter_yamltests/pics_checker.py [hunk 3]
```diff
@@ -53,7 +53,8 @@ def check(self, pics) -> bool:
 
         self.__expression_index = 0
         tokens = self.__tokenize(pics)
-        return self.__evaluate_expression(tokens, self.__pics)
+        if self.__pics is not None:
+            return self.__evaluate_expression(tokens, self.__pics)
```
Comment (vivien-apple) [inline scripts/py_matter_yamltests/matter_yamltests/pics_checker.py:57]: This should not be needed anymore.

**Review by vivien-apple — APPROVED** [review]
Approving assuming the change to `pics_checker.py` will be removed.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**